packages/edit-site/src/components/navigation-inspector/index.js

```diff
@@ -152,39 +58,29 @@ export default function NavigationInspector( { onSelect } ) {
 
 	return (
 		<div className="edit-site-navigation-inspector">
-			{ hasResolvedNavigationMenus && ! hasNavigationMenus && (
-				<p className="edit-site-navigation-inspector__empty-msg">
-					{ __( 'There are no Navigation Menus.' ) }
-				</p>
-			) }
-
-			{ ! hasResolvedNavigationMenus && (
+			{ hasLoadedInnerBlocks &&
+				! isLoadingInnerBlocks &&
+				! navigationBlockId && (
+					<p className="edit-site-navigation-inspector__empty-msg">
+						{ __( 'There are no Navigation Menus.' ) }
+					</p>
+				) }
+
+			{ ! hasLoadedInnerBlocks && (
 				<div className="edit-site-navigation-inspector__placeholder" />
 			) }
-			{ isLoading && (
+			{ isLoadingInnerBlocks && (
 				<>
 					<div className="edit-site-navigation-inspector__placeholder is-child" />
 					<div className="edit-site-navigation-inspector__placeholder is-child" />
 					<div className="edit-site-navigation-inspector__placeholder is-child" />
 				</>
 			) }
-			{ hasInnerBlocks && ! isLoading && (
-				<BlockEditorProvider
```

Comment:
If we're removing the provider from here, where do we get the "blocks" from?

Reply:
Hi @youknowriad, we are now using the main editor instance, and we retrieve the navigation block from there. If we call wp.data.select('core/block-editor).geyBlocks(); we have a tree, and the navigation blocks are there. We are just using it.
We kept the previous logic where the menu shown is associated with the first navigation block or the selected navigation block.

Reply:
That doesn't make a lot of sense to me, the main editor is rendered in the "frame" and its content might change. For instance I can see a website where the home page doesn't contain any menu but you need to get into an internal page first. 

In other words, I still think the inspector is its own block editor instance and we should be able to control what we load in it and not depend on what's shown elsewhere (frame)

Reply:
>That doesn't make a lot of sense to me, the main editor is rendered in the "frame" and its content might change. For instance I can see a website where the home page doesn't contain any menu but you need to get into an internal page first.

If the menu is not on the current frame, how do we know the menu to show? Multiple templates may even reference different navigation menus. If the menu is not in the frame to me, it seems better not to show the navigation option at all, given that we don't have navigation.
The current system using a different block editor does not seem reliable:
- Page lists don't load at all because the edit function is not executed.
- There are many crashes when using two block editor providers where both reference the menu entity. E.g: try to add some menu items on the sidebar, remove, etc; soon the editor crashes.

In the previous system, the one we have on the trunk and that comes from the navigation inspector sidebar days, if there was no menu referenced in the frame just showed the first navigation menu even if no template references it..

I guess we can merge this PR using a single block editor and we use a single editor every time is possible. If we really want to show the navigation editor even if navigation is not in the frame, I can add a follow-up that in that case, loads the first menu as before in a new block editor provider. Given that in that case there are not multiple block-editor providers referencing the same entity I guess we will not have crashes. And doing this we keep the same functionality as before. How does that sound?

Reply:
For clarity, In trunk right now, when the "navigation menus" is shown in the sidebar, we're always showing the home page.

> If the menu is not on the current frame, how do we know the menu to show?

I think that's the main question that we need to solve here.  IMO, the sidebar is higher-level than the frame in other words, the frame should never dictate what's shown in the sidebar, it's the other way around.

From that perspective, Any of the following approaches are good to me:

 - Show the first menu with a "select" to switch which menu to show 
 - Show the first menu with the assumption that most sites only have a single menu (so don't bother with the "select")
 - Have a way (could be a "guideline" or "convention") to know which menu is the "main menu" of the site and always show that one.

Reply:
Hi @youknowriad, I updated the code now we keep the behavior of loading the first navigation menu in a new block editor if it is not in canvas. When it is in canvas we don't load a new block editor which solves many crashes.
When the navigation menu is not in canvas we render a hidden block editor so edit functions are executed and things like the page list load correctly.
With the current state, we don't lose any functionality when compared to trunk.

Reply:
> . When it is in canvas we don't load a new block editor which solves many crashes.

@jorgefilipecosta I think we should always have a separate block editor in that sense, we shouldn't be assuming that a canvas exists in the first place. 

> solves many crashes.

I'd like to understand what crashes it solves, because for me this is an indication that we have bugs elsewhere and we're just hacking our way around it.

Reply:
I'm not being able to reproduce the crashes right now. Previously they were very frequent and easy to replicate something probably changed meanwhile. The crashes were from react saying we have recursive calls to setState.
I will pay attention and see if they come back.

I kept exactly the same logic as before, and reverted back to my first commit where I'm only adding a hidden block editor so edit functions execute and page lists load.

I still think if the navigation block is loaded on the canvas it would be better to not load a new block editor and hidden block list it would be more performant to have a single block editor instead of two. But in order to keep this PR simple and focus on the page list I guess we can go with the hidden block list for now.